### ISSUE TITLE
I couldnt find the file "conda environment yaml"

### ISSUE THREAD

**Issue by Missourl** [wandb/wandb#5378]
Hello,

In the aim of reproducing the best run from Wandb Sweep, I tried to follow this Video "https://www.youtube.com/watch?v=IYtGw1btOsM&t=148s".
The Problem is I cant find the file "conda environment yaml" (See picture)

Could you please help?
![Screenshot 2023-04-19 221116](https://user-images.githubusercontent.com/93824605/233189128-4672d9bb-fd39-46b8-bdf1-928889f5d744.png)

Thank you

**Comment by MBakirWB** [comment]
Hi @Missourl , are you executing your runs from inside a conda environment? Wandb will automatically save your [conda environment](https://github.com/wandb/wandb/blob/cd19b480eed03037c898667fddad0a43fcdd8a3b/wandb/sdk/internal/system/system_info.py#L246)

**Comment by Missourl** [comment]
Ah therefore , indeed no.
Thank your for your answer.
So Is there are possibilities to reproduce the best run beside this file "conda environement yaml" please?

**Comment by nate-wandb** [comment]
Hi @Missourl, are you looking to have the config settings from the best run? The conda yaml would only help with dependencies in the environment but the config of the run would be able to help with model settings.

**Comment by Missourl** [comment]
Hello ,

Sorry for late answer.
No my Problem is not the run setting, I already tried my best run on my local computer and I didnt get the same result as the one in wandb. And when I have seen this video "https://www.youtube.com/watch?v=IYtGw1btOsM&t=148s", I thought it might be the reason. so if I would reformulate my question, what should I consider if I take the same best run setting and try it on my local computer, to get the same result ?

Thanks
Best Regards

**Comment by nate-wandb** [comment]
@Missourl, I see, so are you looking to match the results from the video? Sorry if I'm misunderstanding. There could be other factors such as random seem the training starts with that could affect the results but I just want to make sure I am understanding your goal.

Thank you,
Nate

**Comment by Missourl** [comment]
No, I dont want to match the results from the video. Iam working on my own project, and I want to reproduce the best run  of wandb results on my local computer (see picture, the one with the red circle)
![github](https://github.com/wandb/wandb/assets/93824605/715cae13-d5ac-466f-afda-4661038cc61d)

So please, if I want to reproduce the same result (the one with red circle)  , what should I do.  I already tried the same configuration but I didnt get the same result.

Thanks

**Comment by nate-wandb** [comment]
I see. Is the training data the same and fed to the model in the same order? The only other variable would likely be the random seed used to start the model which would be hard to reproduce if this was not logged in the initial run. 

You can use the requirements.txt saved in the files tab to make sure all of your packages are pinned to the same version as your original experiment as well which I believe is what you were looking to do with the condo.yaml

**Comment by nate-wandb** [comment]
@Missourl I think the requirements.txt will likely not have as much effect on the model as the random seed. That said, I would be pretty surprised if you use the same config, dataset, and model architecture but are still getting dramatically different results from the model.

**Comment by Missourl** [comment]
I will give it another try, but I would like to be sure about some points:  (PS: Iam working on a reinforcement learning problem.)

1)- when you talk about the random seed, do you mean creating the same initial observation and all the others parameters that I need to care about it on my environment ? or do you mean the initial weights and biases  giving to the neural network?

2)- in the run that I would like to reproduce, I have found differents learning rate in differents positions (See Pictures)

 
![pic1](https://github.com/wandb/wandb/assets/93824605/0e9a84f2-2cc0-4f16-a9a8-73d2b5560605)
![pic2](https://github.com/wandb/wandb/assets/93824605/aac663b8-543e-45c0-b945-19d2e0a878d5)
![pic3](https://github.com/wandb/wandb/assets/93824605/2a5aae29-ce98-4727-b053-aa8c27638c4b)

**Comment by nate-wandb** [comment]
@Missourl 
1. I am referring to the latter. The random seed for the starting weights and biases of the model. 
2. I would have to see your code to know for sure. It looks like you set the learning rate in your config but then also are saving it as a metric. If you pass the learning rate to your model with `wandb.config["learning_rate"]` then this would be the learning rate that your model uses.

**Comment by Missourl** [comment]
1)- I didnt set the random seed when I ran it on wandb, so there are really no chance to reproduce the same results?
if  so Iam wondering if I could upload the trained model from wandb and take the weights and biases of NN ? (Iam willing to extend my obs space therefore I need to extend also my NN shape).

2)- for  learning rate , your answer helped me to figure it out. Thanks :)

**Comment by nate-wandb** [comment]
Do you have a model checkpoint and would like to resume training? If so, yes you can upload the checkpoint file to a [W&B Artifact](https://docs.wandb.ai/guides/artifacts) and then use it to spawn new runs to further train.

**Comment by Missourl** [comment]
Ok thanks a lot
